metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -747,6 +786,45 @@ def get_lineage_if_enabled(
                         ).as_workunit()
         return None
 
+    def get_fine_grained_lineages(
+        self,
+        dataset_urn: str,
+        s3_dataset_urn: str,
+        schema_metadata: SchemaMetadata,
+        schema_metadata_for_s3: SchemaMetadata,
+    ) -> Optional[List[FineGrainedLineageClass]]:
+        if schema_metadata and schema_metadata_for_s3:
+            fine_grained_lineages: List[FineGrainedLineageClass] = []
+            for field in schema_metadata.fields:
+                matching_s3_field = next(
+                    (
+                        f
+                        for f in schema_metadata_for_s3.fields
+                        if f.fieldPath.split(".")[-1] == field.fieldPath.split(".")[-1]
```

Comment:
Will this work for nested fields?
For example:
a.b.c != a.c

Reply:
Here is an example fieldPath for the above:
```
unique_v2_field_paths = [
 "[version=2.0].[type=union].[type=A].[type=string].f",
 "[version=2.0].[type=union].[type=B].[type=string].f"
]
```

Reply:
hi @treff7es it will work only for top level as s3 it self does have support for nested level as per my opinion

here is output for nested level 

<img width="1422" alt="Screenshot 2024-07-17 at 5 46 05 PM" src="https://github.com/user-attachments/assets/392d522f-9154-455f-b00b-44007f3b7ce6">

Reply:
hi @treff7es 
I have tested the scenario, and it seems that the lineage will only connect to the top-level fields. The lineage mapping does not work for nested column levels.

For eg 
Glue Schema
The field path in Glue at the nested level is:
 `[version=2.0].[type=struct].[type=union].[type=struct0].union_type.[type=string].mapping` 

S3 Schema
The top-level field path in S3 is:
`"fieldPath": "mapping"`

Nested Field Details
```
{
                    "fieldPath": "union_type",
                    "nullable": false,
                    "type": {
                        "type": {
                            "com.linkedin.schema.RecordType": {}
                        }
                    },
                    "nativeDataType": "struct<int_value: int32, double_value: double, array_value: list<element: string>, struct_value: struct<employee_id: int32, mapping: string>>",
                    "recursive": false,
                    "isPartOfKey": false
 }
```
The lineage is not correctly lineage the field "mapping" because there is no corresponding top-level field in Glue. As a result, the lineage mapping fails.

output of this 

<img width="620" alt="Screenshot 2024-07-18 at 11 18 16 AM" src="https://github.com/user-attachments/assets/79606e4a-6909-4842-a746-3a0d8d87d33e">
<img width="555" alt="Screenshot 2024-07-18 at 11 19 54 AM" src="https://github.com/user-attachments/assets/acd2610e-f401-4c19-b388-757c0d362786">

Reply:
I still think for correctness, we should check if we found the match on the same attribute. 
Why can't we compare the path fully?
If the issue is a v2 vs. v1 field path, then I think we should convert from one to another to be correct.
You can use the following method to convert v2 to v1 for comparison -> https://github.com/datahub-project/datahub/blob/452b94fb023356b7a3392c878a67aed9c6b722cf/metadata-ingestion/src/datahub/api/entities/dataset/dataset.py#L396

Reply:
hi @treff7es yes, issue is related to  v2 vs. v1 field path only i ll use above to verify the lineage

Reply:
hi @treff7es implemented the changes mentioned above

Output:

<img width="543" alt="Screenshot 2024-07-19 at 1 34 43 PM" src="https://github.com/user-attachments/assets/eedb68c0-5fd0-4667-8b6c-b3848548d065">

<img width="504" alt="Screenshot 2024-07-19 at 1 34 51 PM" src="https://github.com/user-attachments/assets/4c94bb91-d5bb-47b0-9b79-877a5cb8ba9e">

Let me know if anything is required

Thanks !!!

Reply:
Thanks. It's much better and safer now. I left one small comment, and I think it will be fine.

Reply:
i have fixed that, Thanks !!!

Reply:
thanks, it looks good now